### PR TITLE
feat(reset password): resets password

### PR DESCRIPTION
**Pull request by James-nyagatare**
## What does this PR do?
-resets password
## Description of Task to be completed?
```
- Sends reset password link to an email
- Resets password 
```
## How should this be manually tested?
- Use swagger documentation 
## Any background context you want to provide?
- N/A
## What are the relevant pivotal tracker stories?
[Relevant story](https://www.pivotaltracker.com/story/show/176153386)
## Screenshots (if appropriate)
-N/A
## Questions:
-N/A

### CODE REVIEW

**Review by unadine — COMMENTED** [review]
all good

**Review by Mucyo-Aime-Christian — COMMENTED** [review]
LGTM except if @harerakalex may request for any other changes

File: src/helpers/actions.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const actionsEnum = {
```
Comment (harerakalex) [inline src/helpers/actions.js:1]: Nice Idea @James-nyagatare

File: src/tests/user.test.js [hunk 2]
```diff
@@ -28,4 +30,21 @@ describe('User related tests:', () => {
     expect(res.status).to.be.equal(401);
     expect(res.body).to.have.property('error');
   });
+  it('should not send a reset link to an invalid email', async () => {
+    const res = await chai.request(app).post('/api/v1/users/forgotPassword').send(mockdata.resetInvalidEmail);
+    expect(res.status).to.be.equal(400);
+  });
+  it('should send a reset link to the user', async () => {
+    const res = await chai.request(app).post('/api/v1/users/forgotPassword').send(mockdata.resetEmail);
```
Comment (harerakalex) [inline src/tests/user.test.js:38]: @James-nyagatare make sure that while testing you don't send an email.

Comment (James-nyagatare) [reply src/tests/user.test.js:38]: Implemented @harerakalex .

File: .nycrc [hunk 3]
```diff
@@ -1 +1 @@
-{"reporter": ["lcov","text"], "exclude": ["src/tests", "src/database"]}
\ No newline at end of file
+{"reporter": ["lcov","text"], "exclude": ["src/tests", "src/database", "src/helpers/mailer/sendMailer.js"]}
```
Comment (harerakalex) [inline .nycrc:1]: I did not mean to exclude it, What you can do make sure when you hit the API of sending resent link email. you don't send an email to the user. So from the test use a valid email and check if you can get the email in your inbox. @James-nyagatare

**Pull request merged**